src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -3135,6 +3135,98 @@ private async Task<(long Index, object Result)> SendToNodeAsync(TransactionOpera
             return (command.Result.RaftCommandIndex, command.Result.Data);
         }
 
+        public async Task WaitForExecutionOnSpecificNode(JsonOperationContext context, string nodeTag, long index)
+        {
+            await WaitForExecutionOnRelevantNodesAsync(context, members: new List<string> { nodeTag }, index);
+        }
+
+        public async Task WaitForExecutionOnRelevantNodesAsync(JsonOperationContext context, List<string> members, long index)
+        {
+            await Cluster.WaitForIndexNotification(index); // first let see if we commit this in the leader
+
+            if (members.Count == 0)
+                throw new InvalidOperationException("Cannot wait for execution when there are no nodes to execute ON.");
+
+            var executors = GetClusterRequestExecutorsForMembers(members);
```

Comment:
I think it is time to have a single cluster request executor and use `SelectedNode` for the commands.
cc @ppekrol 

probably a separate issue though.

Reply:
I did a rebase and see that this is already implemented